backend/src/main/kotlin/app/ehrenamtskarte/backend/stores/importer/steps/StoreToDatabase.kt

```diff
@@ -0,0 +1,65 @@
+package app.ehrenamtskarte.backend.stores.importer.steps
+
+import app.ehrenamtskarte.backend.stores.database.*
+import app.ehrenamtskarte.backend.stores.importer.ImportMonitor
+import app.ehrenamtskarte.backend.stores.importer.PipelineStep
+import app.ehrenamtskarte.backend.stores.importer.types.ImportAcceptingStore
+import org.jetbrains.exposed.dao.id.EntityID
+import org.jetbrains.exposed.sql.deleteAll
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.postgis.Point
+
+class StoreToDatabase(val monitor: ImportMonitor) : PipelineStep<List<ImportAcceptingStore>, Unit> {
+
+    override fun execute(acceptingStores: List<ImportAcceptingStore>) {
+        transaction {
+            try {
+                PhysicalStores.deleteAll()
+                AcceptingStores.deleteAll()
+                Contacts.deleteAll()
+                Addresses.deleteAll()
+
+                for ((done, acceptingStore) in acceptingStores.withIndex()) {
+                    val address = AddressEntity.new {
+                        street = acceptingStore.street
+                        postalCode = acceptingStore.postalCode
+                        locaction = acceptingStore.location
+                        countryCode = acceptingStore.countryCode
+                    }
+                    val contact = ContactEntity.new {
+                        email = acceptingStore.email
+                        telephone = acceptingStore.telephone
+                        website = acceptingStore.website
+                    }
+                    val storeEntity = AcceptingStoreEntity.new {
+                        name = acceptingStore.name
+                        description = acceptingStore.discount
+                        contactId = contact.id
+                        categoryId = EntityID(acceptingStore.categoryId, Categories)
+                    }
+                    PhysicalStoreEntity.new {
+                        storeId = storeEntity.id
+                        addressId = address.id
+                        coordinates = Point(acceptingStore.longitude, acceptingStore.latitude)
+                    }
+                    drawSuccessBar(done, acceptingStores.size)
+                }
+            } catch (e: Exception) {
+                monitor.addMessage("Unknown exception while storing to db", e)
+                rollback()
+                throw e
+            }
+        }
+    }
+
+    private fun drawSuccessBar(done: Int, of: Int) {
+        if (done % 200 == 0) {
+            var bar = ""
+            repeat(done / 200) { bar += "##" }
+            var remaining = ""
+            repeat((of - done) / 200) { remaining += "__" }
+            println("[$bar$remaining]")
```

Comment:
Just a trifle: I have read one can replace lines on the console using carriage return (https://stackoverflow.com/a/11777847/3987396)